shoes-swt/lib/shoes/swt/generate_backend.rb

```diff
@@ -1,7 +1,35 @@
 require 'rbconfig'
 
-def generate_backend(path)
-  options = "-J-XstartOnFirstThread" if RbConfig::CONFIG["host_os"] =~ /darwin/
+class Shoes
+  module SelectedBackend
+    class << self
+      def validate
+        if RUBY_ENGINE != "jruby"
+          puts <<-EOS
 
-  "jruby #{options} #{path}/shoes-swt"
+*******************************************************************************
+The shoes-swt backend requires a 9.X version of JRuby.
+
+You are running the following Ruby instead:
+
+  #{RUBY_DESCRIPTION}
+
+Please download JRuby from http://jruby.org (or your favorite Ruby version
+manager), reinstall the `shoes` gem on that Ruby, and try again.
+
+Sorry for the inconvenience!
```

Comment:
Linking to our material's a great idea. Pushed up a revision... what do you think?